addon-test-support/-private/mock-window.js

```diff
@@ -4,8 +4,12 @@ export function mockWindow(scope) {
   if (!scope) {
     throw new Error('mockWindow must be called with `this` as the first function parameter!');
   }
-  if (!scope.register) {
-    throw new Error('mockWindow must be called from an integration test!');
+
+  if (scope.owner && scope.owner.register) {
+    return scope.owner.register('service:window', windowMockFactory(), { instantiate: false });
```

Comment:
Ah, I see... I think the previous error was that we always called the `register` function with `scope` as `this`, but in this case it is actually `scope.owner`. At least that's only real difference I see here...